app/platform/platform.c

```diff
@@ -859,3 +859,8 @@ uint32_t platform_flash_mapped2phys (uint32_t mapped_addr)
   uint32_t meg = (b1 << 1) | b0;
   return mapped_addr - INTERNAL_FLASH_MAPPED_ADDRESS + meg * 0x100000;
 }
+
+void* platform_print_deprecation_note( const char *msg, const char *time_frame)
```

Comment:
It depends on much of a nuisance we want to make of our deprecation notes. I could even see an argument made for a two-stage approach, where initially we only whinge once, but closer to the date we keep whinging. Quite possibly over-kill though...

If you want to keep track of messages already printed, the easy option would be a singly-linked list which stores the msg pointer as the value. If you're wanting to be really tight you could probably use less than the standard 32bits, but that would not be warranted unless we have lots of deprecations in flight, which I hope we won't.